pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -2654,6 +2654,36 @@ func updatePortAndPciBackIoBundleAll(ctx *domainContext) {
 	}
 }
 
+// nvmeDeviceIsMountedByEVE checks if an NVME device is used by EVE-OS
+func nvmeDeviceIsMountedByEVE(pciID string) bool {
+	// Find /dev/nvmX from pciID
+	// e.g., ls /sys/bus/pci/devices/<pciID>/nvme/
+	//  -> nvme1
+	cmd := "/bin/ls"
+	cmdArgs := []string{"/sys/bus/pci/devices/" + pciID + "/nvme/"}
+	out, err := base.Exec(log, cmd, cmdArgs...).CombinedOutput()
+	if err != nil || out == nil {
+		log.Errorf("Could not find device with pciID %s, error:%+v, out:%s", pciID, err, string(out))
+		return false
+	}
+
+	// Extract nvmX from previous output
+	// e.g., out is /sys/class/nvme/nvme0/device/uevent:PCI_SLOT_NAME=0000:00:04.0
+	//  -> get nvme0 which is the 5th token of out using "/" delimiter
+	deviceName := strings.TrimSpace(string(out))
+	cmd = "/bin/grep"
+	cmdArgs = []string{deviceName, "/proc/mounts"}
+	out, err = base.Exec(log, cmd, cmdArgs...).CombinedOutput()
+	// If grep did not succeed, then device is not used
+	if err != nil || out == nil {
+		log.Noticef("Device %s is not used by EVE", deviceName)
+		return false
+	}
+
+	log.Noticef("Device with pciID %s is used by EVE, error:%+v, out:%s", pciID, err, string(out))
```

Comment:
Will this function be called periodically? If so there would be lots of log entries being sent from 1M devices.